core/src/main/java/org/hisp/dhis/android/network/legendset/LegendSetDTOMapper.kt

```diff
@@ -36,15 +37,16 @@ internal fun legendSetDtoToDomainMapper(item: LegendSetDTO): LegendSet {
     return LegendSet.builder()
         .applyBaseIdentifiableFields(item)
         .symbolizer(item.symbolizer)
-        .legends(item.legends.map { legendDtoToDomainMapper(it) })
+        .legends(item.legends.map { legendDtoToDomainMapper(it, item.uid) })
         .build()
 }
 
-internal fun legendDtoToDomainMapper(item: LegendDTO): Legend {
-    return Legend.builder()
-        .applyBaseIdentifiableFields(item)
-        .startValue(item.startValue)
-        .endValue(item.endValue)
-        .color(item.color)
-        .build()
+internal fun legendDtoToDomainMapper(item: LegendDTO, legendSetUid: String? = null): Legend {
+    return Legend.builder().apply {
+        applyBaseIdentifiableFields(item)
+        legendSetUid?.let { legendSet(ObjectWithUid.create(legendSetUid)) }
```

Comment:
In the `main` code, this legendSetUid can't be empty. Actually, it comes from a `BaseIdentifiableObjectDTO` whose `uid` property is not nullable. The reason to make it nullable here is the test, isn't it? Couldn't we make it non-null in this method and just pass any string in the test case?